File: src/main/java/es/qky/selenidepom/Required.java
package es.qky.selenidepom;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.ElementType;


/**
 * Every field with @Required annotation in a Page or a Widget will be checked if visible when shouldLoadRequired method is called.
 */
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Target({ElementType.FIELD, ElementType.METHOD})
public @interface Required {
}

File: src/main/kotlin/com/github/qky666/selenidepom/annotation/Required.kt
package com.github.qky666.selenidepom.annotation

/**
 * Every field, property or method (without parameters) annotated in a [com.github.qky666.selenidepom.pom.Page]
 * subclass will be checked if visible when [com.github.qky666.selenidepom.pom.Page.shouldLoadRequired] or
 * [com.github.qky666.selenidepom.pom.Page.hasLoadedRequired] is called.
 *
 * @property value Represents 'when' the element is required.
 * Using default value means the element is always checked, regardless of pomVersion parameter
 * passed to [com.github.qky666.selenidepom.pom.Page.shouldLoadRequired] or
 * [com.github.qky666.selenidepom.pom.Page.hasLoadedRequired].
 * You can set a value like "desktop" or "mobile" to represent that the web element
 * ([com.codeborne.selenide.SelenideElement], [com.github.qky666.selenidepom.pom.Widget], etc.)
 * annotated is only required in the desktop or mobile version respectively of the web page.
 */
@MustBeDocumented
@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FUNCTION, AnnotationTarget.FIELD)
@Repeatable
annotation class Required(val value: String = "")
